leetcode-extension/background.js

```diff
@@ -15,130 +15,112 @@ query {
 
 // this function will retry for 3 times if any error occur while fetching the leetcode graphql
 let fetchLeetcodeData = async () => {
-  let retries = 3;
-  let success = false;
-  while (retries > 0 && !success) {
-    try {
-      // console.log(`Fetching daily coding challenge from LeetCode API.`)
-      const init = {
-        method: "POST",
-        headers: { "Content-Type": "application/json" },
-        body: JSON.stringify({ query: LEETCODE_GRAPHQL_QUERY }),
-      };
+    let retries = 3;
+    let success = false;
+    while (retries > 0 && !success) {
+        try {
+            // console.log(`Fetching daily coding challenge from LeetCode API.`)
+            const init = {
+                method: "POST",
+                headers: {"Content-Type": "application/json"},
+                body: JSON.stringify({query: LEETCODE_GRAPHQL_QUERY}),
+            };
 
-      const response = await fetch(LEETCODE_API_ENDPOINT, init);
-      success = true;
-      return response.json();
-    } catch (error) {
-      console.log(`Error: ${error}. Retrying...`);
-      retries--;
+            const response = await fetch(LEETCODE_API_ENDPOINT, init);
+            success = true;
+            return response.json();
+        } catch (error) {
+            console.log(`Error: ${error}. Retrying...`);
+            retries--;
+        }
     }
-  }
 
-  if (!success) {
-    console.log("Failed to call API after 3 retries");
-  }
+    if (!success) {
+        console.log("Failed to call API after 3 retries");
+    }
 };
-const domainForWhitelist = new Set(["leetcode.com","accounts.google.com","extensions","github.com"]); // this set is to whitelist the redirection for chrome://extensions and accounts.google.com
+
+function redirect(url) {
+    chrome.tabs.query(
+        {currentWindow: true, active: true},
+        (tabs) => {
+            const currURL = new URL(tabs[0].url);
+            const domain = currURL.hostname;
+
+            if (!domain.includes("leetcode") && !domainWhiteList.has(domain)) {
+                chrome.tabs.update({url: url});
+            }
+        }
+    );
+}
+
+const domainWhiteList = new Set(["leetcode.com", "accounts.google.com", "extensions", "github.com"]); // this set is to whitelist the redirection for chrome://extensions and accounts.google.com
 // this function will redirect to leetcode.com
 function leetcodeForcer() {
-  let leetcodeData = fetchLeetcodeData();
```

Comment:
As I had already fixed that bug yesterday, from this commit what I can see is you had generalised the redirection code force every scenario ? Correct me if I am wrong

Reply:
Yes. That is exactly what I did there

Reply:
However, it is important to note that we are talking about slightly different bugs, I believe. You added the whitelist logic to prevent redirection from important pages like the extensions page. However, it was only set in one of the control flows: When user is not signed in (`data !== undefined && !data.data.userStatus.isSignedIn`). This meant the whitelist was not working when user is signed in and only whitelisting leetcode.com

Reply:
Got it I forgot for that workflow thanks for this change.